packages/nx-python/src/executors/build/executor.ts

```diff
@@ -88,14 +72,33 @@ export default async function executor(
       readFileSync(buildPyProjectToml).toString('utf-8')
     ) as PyprojectToml;
 
-    logger.info(chalk`  Resolving dependencies...`);
-    resolveLockedDependencies(
+    const deps = resolveDependencies(
+      options,
       root,
       buildFolderPath,
       buildTomlData,
-      options.devDependencies,
-      workspaceRoot
+      workspaceRoot,
+      context
     );
+
+    const pythonDependency = buildTomlData.tool.poetry.dependencies.python;
+
+    buildTomlData.tool.poetry.dependencies = {};
+    buildTomlData.tool.poetry.group = {
+      dev: {
+        dependencies: {},
+      },
+    };
```

Comment:
After more testing I have found this is an issue. I think all dependency groups should be handled in the same way as the main group. The only change that should be made with regards to the dependencies is to re-write the local ones to instead use the package name, or name and source, as previously discussed.

Reply:
I'm new to Poetry and it looks like I've been confusing extras with groups. I'll look into extras and see if that makes sense to use instead.

Reply:
@niall-mediasuite when poetry generates the `.tar.gz` only the main dependencies are listed in the `setup.py`, that's why in the build executor I'm only looking at the main dependencies section, 

you can prove this behavior by creating a poetry project, adding some main dependencies, adding some dependencies to a different group, and running the `poetry build` command.

Then if you look at the `setup.py` stored in the `.tar.gz` file, only the main dependencies will be listed there.

Reply:
Ah, good to know. Thanks, I'll have a play around with them.